proteuslib/unit_models/tests/test_reverse_osmosis_1D.py

```diff
@@ -293,4 +565,843 @@ def test_solution(self, RO_frame):
+
+        m.fs.unit.feed_inlet.flow_mass_phase_comp[0, 'Liq', 'NaCl'].fix(
+            feed_flow_mass * feed_mass_frac_NaCl)
+
+        m.fs.unit.feed_inlet.flow_mass_phase_comp[0, 'Liq', 'H2O'].fix(
+            feed_flow_mass * feed_mass_frac_H2O)
+
+        m.fs.unit.feed_inlet.pressure[0].fix(feed_pressure)
+        m.fs.unit.feed_inlet.temperature[0].fix(feed_temperature)
+        m.fs.unit.A_comp.fix(A)
+        m.fs.unit.B_comp.fix(B)
+        m.fs.unit.permeate_outlet.pressure[0].fix(pressure_atmospheric)
+        m.fs.unit.length.fix(8)
+        m.fs.unit.recovery_vol_phase[0, 'Liq'].fix(0.4)
+        m.fs.unit.cp_modulus.fix(1.1)
+
+        # test ports
+        port_lst = ['feed_inlet', 'feed_outlet', 'permeate_outlet']
+        for port_str in port_lst:
+            port = getattr(m.fs.unit, port_str)
+            assert len(port.vars) == 3  # number of state variables for NaCl property package
+            assert isinstance(port, Port)
+
+        # test pyomo objects on unit
+        unit_objs_type_dict = {'dens_solvent': Param,
+                               'A_comp': Var,
+                               'B_comp': Var,
+                               'flux_mass_phase_comp': Var,
+                               'area': Var,
+                               'area_cross': Var,
+                               'width': Var,
+                               'length': Var,
+                               'recovery_vol_phase': Var,
+                               'cp_modulus': Var,
+                               'mass_transfer_phase_comp': Var,
+                               'eq_mass_transfer_term': Constraint,
+                               'eq_permeate_production': Constraint,
+                               'eq_flux_mass': Constraint,
+                               'eq_connect_mass_transfer': Constraint,
+                               'eq_feed_isothermal': Constraint,
+                               'eq_permeate_isothermal': Constraint,
+                               'eq_recovery_vol_phase': Constraint,
+                               'eq_area': Constraint,
+                               'eq_mass_flux_equal_mass_transfer': Constraint,
+                               'eq_permeate_outlet_isothermal': Constraint,
+                               'eq_permeate_outlet_isobaric': Constraint
+                               }
+        for (obj_str, obj_type) in unit_objs_type_dict.items():
+            obj = getattr(m.fs.unit, obj_str)
+            assert isinstance(obj, obj_type)
+        # check that all added unit objects are tested
+        for obj in m.fs.unit.component_objects(
+                [Param, Var, Expression, Constraint], descend_into=False):
+            obj_str = obj.local_name
+            if obj_str[0] == '_':
+                continue  # do not test hidden references
+            assert obj_str in unit_objs_type_dict
+
+        # test feed-side control volume and associated stateblocks
```

Comment:
We don't need to do this for each case, I'd get rid of this for all but the first one.